doc/proposals/images-subspec.md

```diff
@@ -0,0 +1,257 @@
+
+Follow a kubernetes api pattern and provide an ImageSpec and ImageListSpec structs that
+operator CRDs can reuse. I imagine operator-sdk would publish these like the
+[PodSpec](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2831) found for kubernetes.
+
+OpenAPIv3 Def for ImageSpec:
+
+```yaml
+type: Object
+properties:
+  name:
+    type: string
+    description: Name of image used for lookup.
+  image:
+    type: string
+    description: Location of the image.
+  pullPolicy:
+    type: string
+    description: Pull Policy for the image.
+  pullSecrets:
+    type: array
+    description: Array of pull secret names to use.
+    items:
+      types: string
+required: ['name', 'image']
+```
+
+OpenAPIv3 Def for ImageListSpec:
+
+```yaml
+images:
+  type: array
+  items:
+    type: object
+    properties:
+      name:
+        type: string
+        description: Name of image used for lookup.
+      image:
+        type: string
+        description: Location of the image.
+      pullPolicy:
+        type: string
+        description: Pull Policy for the image.
+      pullSecrets:
+        type: array
+        description: Array of pull secret names to use.
+        items:
+          types: string
+  required: ['name', 'image']
+```
+
+#### Add an image flag to sdk add api
+
+Add a flag to preset images for operator-sdk add api command.
+
+```bash
+operator-sdk add api --api-version=app.example.com/v1alpha1 \
+  --kind=AppService --image=ngnix:latest
+operator-sdk add api --api-version=app.example.com/v1alpha1 \
+  --kind=AppService \
+  --image=ngnix:latest \
+  --image=redis:latest
+```
+
+This flag will create an AppServiceSpec struct with prefilled data.
+
+```golang
+type AppServiceSpec struct {
+
+  // Object map of image to name to an image location, pull
+  // secret names, and pull policy for the image.
+  Images operatorsdk.ImagesSpec
+
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+}
+```
+
+#### Add a generate flag for operator-sdk specs
+
+Generate flag added to operator-sdk add controller.
+
+**Blanket generate:**
+```bash
+ operator-sdk add controller --api-version=app.example.com/v1alpha1 \
+   --kind=AppService  \
+   --generate
+```
+
+The generate flag will create the controller with scaffolding using the Images
+property to create resources to be deployed. The generate command could be a
+global flag that enables any type of operatorsdk features that generate
+controller code. And we can include an option to subselect a generation.
+
+**Subselecting a generation:**
+```bash
+ operator-sdk add controller --api-version=app.example.com/v1alpha1 \
+   --kind=AppService  \
+   --generate=images
+```
+
+### Future Work
+
+#### Validation
```

Comment:
If we scaffold the code all the other features will work as expected and we do not need to add any specific test in the scorecard. IMHO the scorecard should be 100% independent and be allowed to be used in any operator project done with SDK or not. It should ensure the quality of the project not matter if the user did it by using any feature or manually.

Reply:
I think your scaffold idea moves away from the original proposal and I disagree.

Reply:
The idea of a scorecard to me is to grade you on how well you have done something. Getting a grade on your operator should include following good practices and standardizations. I think looking for this feature should totally be in scope.